### ISSUE TITLE
The http-client cloud create an error with a message or recycle the error object

### ISSUE THREAD

**Issue by fungiboletus** [Vermonster/fhir-kit-client#99]
The http-client creates new errors with missing messages.

While the message argument in the Error constructor is optional, [as you can see in the MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error), it could be nice to include an error message.

Perhaps the error returned by the `request` module can be used a bit more, to reuse the same message or the same error.

In my use case I'm using Sentry to report errors and they appear empty. Having a meaningful message or reusing the error created by `request` would improve the error reporting. What do you think about that ?

https://github.com/Vermonster/fhir-kit-client/blob/439ff3b079f32e21311d863d0c8aff37a6933b95/lib/http-client.js#L8-L26

to something like: 

```javascript
function responseError({ 
   status, 
   rawData, 
   httpVerb, 
   url, 
   headers,
   originalError,
 }) { 
   const error = originalError || new Error(`HTTP Error ${status}`); 
   let data; 
   try { 
     data = JSON.parse(rawData); 
   } catch (e) { 
     data = rawData; 
   } 
   error.response = { status, data }; 
   error.config = { method: httpVerb, url, headers }; 
   logRequestError(error); 
   return error; 
 } 
```

**Comment by Jammjammjamm** [comment]
I agree and will look into this.

**Comment by Jammjammjamm** [comment]
Addressed by #102
